File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_044)_In_Two_Array_Find_Max_Occuring_Element.dart
int findMaxOccuringElement(List<int> L, List<int> R) {
  Map<int, int> map = {};
  for (int i = 0; i < L.length; i++) {
    if (map.containsKey(L[i]))
      map[L[i]]! + 1;
    else
      map[L[i]] = 1;
  }
  int max = 0;
  int maxElement = 0;
  for (int i = 0; i < R.length; i++) {
    if (map.containsKey(R[i])) {
      if (map[R[i]]! > max) {
        max = map[R[i]]!;
        maxElement = R[i];
      }
    }
  }
  return maxElement;
}

void main() {
  var L = [1, 2, 3], R = [3, 5, 7];
  print(
      'Maximum Occuring Element in both Array = ${findMaxOccuringElement(L, R)}');
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_044)_In_Two_Array_Find_Max_Occuring_Element.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int maxOcc(int L[], int R[], int n)
{
	int arr[1000];
	memset(arr, 0, sizeof(arr));
	for (int i = 0; i < n; i++)
	{
		arr[L[i]]++;
		arr[R[i] + 1]--;
	}
	int maxm = arr[0], res = 0;
	for (int i = 1; i < 1000; i++)
	{
		arr[i] += arr[i - 1];
		if (maxm < arr[i])
		{
			maxm = arr[i];
			res = i;
		}
	}
	return res;
}

int main()
{
	int L[] = {1, 2, 3}, R[] = {3, 5, 7}, n = 3;
	cout << "Maximum Occuring Element in both Array = " << maxOcc(L, R, n) << endl;
}
